README.md

```diff
@@ -1,18 +1,180 @@
+   
+   ```json
+    {
+        "Topics": {
+            "activities-eg": [
+                "http://host.docker.internal:6000/webhook-200",
+                "http://host.docker.internal:6000/webhook-400",
+                "http://host.docker.internal:6000/webhook-missing"
+            ],
+            "comments-eg": [
+                "http://host.docker.internal:6000/webhook-404",
+                "http://host.docker.internal:6000/webhook-401",
+                "http://host.docker.internal:6000/webhook-slow-200"
+            ]
+        }
+    }
+   ```
+
+## How it works
+
+The following diagram shows how components interact with each other.
 
+``` plantuml
+@startuml
+' https://github.com/plantuml-stdlib/Azure-PlantUML
+!include <azure/AzureCommon.puml>
+!include <azure/AzureSimplified.puml>
+!include <azure/Compute/AzureAppService.puml>
+!include <azure/Integration/AzureEventGrid.puml>
+!include <azure/Containers/AzureContainerRegistry.puml>
+left to right direction
+frame "Host Computer" {
+	AzureAppService(pub, "Publisher", "")
+	AzureAppService(sub, "Subscriber", "")
+	frame "Docker Desktop" as docker {
+		AzureEventGrid(ege, "Event Grid Emulator", "")
+	}
+}
+frame Azure {
+    AzureContainerRegistry(acr, "Container Registry", "")
+}
+note top of ege
+    /app/appsetings.json
+end note
+
+pub --> ege : "https://localhost:6500"
+sub <-- ege : "http://host.docker.internal:6000"
+ege <-- acr : "docker pull"
+@enduml
+```
+
+- Docker compose pull and start an instance of Event Grid Emulator in Docker Desktop.
+- Emulator uses ```/app/appsettings.json``` to simulate topics registrations.
+- Publisher send event to emulator via ```https://localhost:6500```.
+- Subscriber recive the event notification via ```http://host.docker.internal``` which is automatically resolved to the host ip.
+
+## appsettings.json file format
+
+Here is the schema of the `appsettings.json` file.
+
+In `appsettings.json` file you can add `Topics` section. This section will contain topics registrations and subscribers. For each topics you can define many subscribers as an array of strings. Here is a sample file that define 2 subscribers on each of the 2 topics:
+
+``` json
+{
+    "Topics": {
+        "topic1": [
+            "http://host.docker.internal:6000/subscriber1",
+            "http://host.docker.internal:6000/subscriber2"
+        ],
+        "topics2": [
+            "http://host.docker.internal:6000/subscriber3",
+            "http://host.docker.internal:6000/subscriber4"
+        ]
+    }
+}
+```
+
+To define a subscriber endpoint you simply have to respond to http calls. Here is the simplest .net subscriber endpoint defined using simple API:
+
+``` csharp
+var app = WebApplication.CreateBuilder(args).Build();
+
+app.MapPost("/subscriber1", () => Results.Ok());
+
+app.Run();
+```
+
+
+
+## Building
+
+The project can be built by running `Build.ps1`.
```

Comment:
The `Build.ps1` script is irrelevant here. It contents is related to building a nuget package, not an application. I believe we can remove the script as well as this README section